tests/core/test_docstring_utils.py

```diff
@@ -0,0 +1,237 @@
+    """
+    Test that a docstring without a description raises a ValueError.
+    """
+    docstring = """
+    Args:
+        x: The input value.
+    Returns:
+        int: The input value incremented by one.
+    """
+    with pytest.raises(ValueError, match="Function description is missing"):
+        parse_docstring(docstring)
+
+
+def test_parse_docstring_single_line_single_param():
+    """
+    Test parsing a simple docstring with a single-line description and one parameter.
+    """
+    docstring = """
+    Add one to the input.
+
+    Args:
+        x: The input value.
+    Returns:
+        int: The input value incremented by one.
+    """
+    expected = DocstringInfo(
+        description="Add one to the input.",
+        params={"x": "The input value."},
+        returns="int: The input value incremented by one.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_multiple_params():
+    """
+    Test parsing a docstring with multiple parameters.
+    """
+    docstring = """
+    Calculate the area of a rectangle.
+
+    Args:
+        width: The width of the rectangle.
+        height: The height of the rectangle.
+    Returns:
+        float: The area of the rectangle.
+    """
+    expected = DocstringInfo(
+        description="Calculate the area of a rectangle.",
+        params={"width": "The width of the rectangle.", "height": "The height of the rectangle."},
+        returns="float: The area of the rectangle.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_multi_line_param_description_with_colon():
+    """
+    Test parsing a docstring where a parameter description spans multiple lines and includes a colon.
+    """
+    docstring = """
+    Add one to the input.
+
+    Args:
+        x: The input value.
+            For example: 4
+    Returns:
+        int: The input value incremented by one.
+    """
+    expected = DocstringInfo(
+        description="Add one to the input.",
+        params={"x": "The input value. For example: 4"},
+        returns="int: The input value incremented by one.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_no_params():
+    """
+    Test parsing a docstring with no parameters.
+    """
+    docstring = """
+    Get the current timestamp.
+
+    Returns:
+        str: The current timestamp in ISO format.
+    """
+    expected = DocstringInfo(
+        description="Get the current timestamp.",
+        params={},
+        returns="str: The current timestamp in ISO format.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_google_style():
+    """
+    Test parsing a Google-style docstring.
+    """
+    docstring = """
+    Calculate the sum of two numbers.
+
+    Args:
+        a (int): The first number.
+        b (int): The second number.
```

Comment:
```suggestion
        a (int): The first number.

        b (int): The second number.
```

what would happen if there is a blank line between a and b?

Reply:
this worked, but added tests that cover this